src/builder/styled_str.rs

```diff
@@ -28,38 +28,56 @@ impl StyledStr {
         AnsiDisplay { styled: self }
     }
 
-    pub(crate) fn header(&mut self, msg: impl Into<String>) {
+    /// Display string as header.
+    pub fn header(&mut self, msg: impl Into<String>) {
```

Comment:
`StyledStr`'s details can't be made public until after
- https://github.com/clap-rs/clap/issues/3234
- https://github.com/clap-rs/clap/issues/1433
- https://github.com/clap-rs/clap/issues/3108

At that point, we'll have a better idea of what can and should be public

Reply:
:+1: 

Though, at least in my understanding of how StyledStr gets used, having a public interface would be a prerequsite for #1433 and #3108.

Would it be helpful to you for me to write up a separate issue that pulls out my random scattered comments about `StyledStr` above (and makes them a lot more cohesive)?